plugins/language_typst.lua

```diff
@@ -4,153 +4,213 @@ local syntax = require "core.syntax"
+		["while"]    = "keyword",
+		["break"]    = "keyword",
+		["continue"] = "keyword",
+		["return"]   = "keyword",
+		["as"]       = "keyword",
+		["or"]       = "keyword",
+		["and"]      = "keyword",
+		["not"]      = "keyword",
+
+		["true"]     = "literal",
+		["false"]    = "literal",
+	}
+}
 
-    },
+table.insert(typst_script.patterns, 1, { pattern = { "%(", "%)" }, syntax = typst_script })
+table.insert(typst_script.patterns, 1, { pattern = { "{",  "}"  }, syntax = typst_script })
+
+
+
+local typst_syntax = {
+	name = "Typst",
+	files = { "%.typ$" },
+	comment = "//",
+	block_comment = { "/*", "*/" },
+
+	patterns = {
+		{
+			pattern = { "#()import", "\n"    },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+		{
+			pattern = { "#()include", "\n"   },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+		{
+			pattern = { "#()set", "\n"       },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+		{
+			pattern = { "#()let", "\n"       },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+		{
+			pattern = { "#()show", "\n"      },
+			type    = { "literal", "keyword" },
+			syntax  = typst_script
+		},
+
+		{ pattern = { "#", "%s"},               type = "literal", syntax = typst_script },
+		{ pattern = { "%$", "%$", "\\" },       type = "literal", syntax = typst_math   },
+
+		{ pattern = { "```js", "```", "\\" },   type = "string", syntax = ".js"         },
+		{ pattern = { "```py", "```", "\\" },   type = "string", syntax = ".py"         },
+		-- Possibility to add more syntax later
+		{ pattern = { "`", "`", "\\" },         type = "string"                         },
+
+		{ pattern = "//.*",                     type = "comment"                        },
+		{ pattern = { "/%*", "%*/" },           type = "comment"                        },
+
+		--{ pattern = {"link"},                      type = "typst_underline"                }, -- add underline text later
+
+		-- Finish later
+		{ pattern = "^%s*=+%s.+%f[\n]",         type = "keyword"                        }, -- Bold
+		{ pattern = {"%*_", "_%*"},             type = "keyword"                        }, -- Bold Italic
+		{ pattern = {"_%*", "%*_"},             type = "keyword"                        }, -- Bold Italic
+		{ pattern = {"_", "_"},                 type = "keyword"                        }, -- Italic
+		{ pattern = {"%*", "%*"},               type = "keyword2"                       }, -- Bold
+
+		{ pattern = "[%+%-\\/]",                type = "operator"                       },
+		{ pattern = "\\[nrt]",                  type = "string"                         },
+		{ pattern = "\\u%b{}",                  type = "string"                         },
+		{ pattern = "\\.",                      type = "normal"                         },
+
+	},
+
+	symbols = {}
+}
 
+table.insert(typst_syntax.patterns, 1, { pattern = { "%[", "%]" }, syntax = typst_syntax })
```

Comment:
In this case you could probably get away with adding it at the top, in the actual table, and just use `syntax = ".typ"`.